torch/csrc/jit/passes/onnx/remove_inplace_ops_for_onnx.cpp

```diff
@@ -567,256 +370,390 @@ Value* findArgumentAsInputParam(
+  // and replace SetAttr with inplace op, updating new value onto first seen
+  // alias.
+  replaceAttrWithInplaceOps(block, attr_name_value_map, attr_node_fullname_map);
+}
+
+// Convert inplace ops to outplace version, and record the associated new alias in ValueTracker.
+void InplaceConverter::convertInplaceOpsAndTrackAlias(Block* block) {
   for (auto it = block->nodes().begin(); it != block->nodes().end();) {
     Node* n = *it;
     it++; // node n can be destroyed
 
-    if (nullptr != module_ &&
-        (n->kind() == prim::GetAttr || n->kind() == prim::SetAttr)) {
-      Module moduleClone = (*module_);
-      trackAndRegisterAttributesInBlocks(
-          n,
-          graph,
-          moduleClone,
-          allAttrValues,
-          setAttrValues,
-          nextSetAttrValues);
-    } else if (n->kind() == aten::copy_) {
-      PrepareCopyForONNX(n);
-    } else if (n->kind() == aten::index_put || n->kind() == aten::index_put_) {
-      PrepareIndexPutForONNX(n);
-    } else if (mr.inplaceOpVariant(n)) {
-      PrepareInplaceOpsInBlocksForONNX(n);
-    } else if (n->kind() == aten::pop) {
-      PrepareListPopForONNX(n);
-    } else if (n->kind() == aten::insert || n->kind() == aten::append) {
-      PrepareListAppendAndInsertForONNX(n);
-    } else if (n->kind() == aten::Delete) {
-      PrepareListDeleteForONNX(n);
-    } else { // for prim::If and prim::Loop nodes with blocks.
+    auto nkind = n->kind();
+    if (nkind == prim::If || nkind == prim::Loop) {
       for (Block* sub_block : n->blocks()) {
-        std::unordered_map<std::string, Value*> map_ =
-            registerInplaceOpAsBlockOutputs(
-                sub_block, graph, allAttrValues, setAttrValues, mr, module_);
-        std::unordered_map<std::string, Value*>::iterator mapIt;
-        for (mapIt = map_.begin(); mapIt != map_.end(); mapIt++) {
-          setAttrValues[mapIt->first] = mapIt->second;
+        convertInplaceOpsAndTrackAlias(sub_block);
+      }
+    } else {
+      Value *orig_data = nullptr, *new_out = nullptr;
+      if (nkind == aten::copy_) {
+        std::tie(orig_data, new_out) = PrepareCopyForONNX(n);
+      } else if (nkind == aten::index_put || nkind == aten::index_put_) {
+        std::tie(orig_data, new_out) = PrepareIndexPutForONNX(n);
+        if (nkind == aten::index_put) {
+          // special case, index_put is not inplace.
+          continue;
         }
+      } else if (nkind == aten::insert || nkind == aten::append) {
+        std::tie(orig_data, new_out) = PrepareListAppendAndInsertForONNX(n);
+      } else if (mr_->inplaceOpVariant(n)) {
+        std::tie(orig_data, new_out) = PrepareInplaceOpsInBlocksForONNX(n);
+      } else if (nkind == aten::pop) {
+        std::tie(orig_data, new_out) = PrepareListPopForONNX(n);
+      } else if (nkind == aten::Delete) {
+        std::tie(orig_data, new_out) = PrepareListDeleteForONNX(n);
+      } else if (nkind == aten::_set_item) {
```

Comment:
I see all the setAttr and getAttr have been converted to inplace ops, I was wondering what exactly is aten::_set_item

Reply:
created by `tensor_list[i] = some_tensor`